testing/dart/codec_test.dart

```diff
@@ -28,10 +28,11 @@ void main() {
 
   test('Fails with invalid data', () async {
     final Uint8List data = Uint8List.fromList(<int>[1, 2, 3]);
-    expect(
-      () => ui.instantiateImageCodec(data),
-      throwsA(exceptionWithMessage('Invalid image data'))
-    );
+    try {
+      await ui.instantiateImageCodec(data);
+    } on Exception catch (e) {
+      expect(e.toString(), contains('Invalid image data'));
+    }
```

Comment:
```suggestion
    late Exception exception;
    try {
      await ui.instantiateImageCodec(data);
    } on Exception catch (e) {
      exception = e;
    }
    expect(exception.toString(), contains('Invalid image data'));
```

Otherwise this will no longer assert that an exception actually gets thrown.

Reply:
Or you could do it like below, with a `fail` immediate after the `await ui.instnatiateImageCodec`.

Reply:
Used a `fail()`.